crates/storage/codecs/src/lib.rs

```diff
@@ -170,25 +177,27 @@ where
     T: Compact,
 {
     /// Returns 0 for `None` and 1 for `Some(_)`.
+    #[inline]
     fn to_compact<B>(self, buf: &mut B) -> usize
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
-        let mut tmp = Vec::with_capacity(64);
+        let Some(element) = self else {
+            return 0;
+        };
 
-        if let Some(element) = self {
-            // We don't know the length until we compact it
-            let length = element.to_compact(&mut tmp);
+        // We don't know the length of the element until we compact it.
+        let mut tmp = Vec::with_capacity(64);
```

Comment:
would appreciate a comment on why `64`

Reply:
Don't know either, I didn't write this code